index.ts

```diff
@@ -0,0 +1,154 @@
+// Neara Infrastructure Task
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+const stack = pulumi.getStack();
+const tags = { Name: "neara-task" };
+
+// Create VPC and subnets
+const vpc = new aws.ec2.Vpc("andy-vpc", {
+  cidrBlock: "10.0.0.0/16",
+  tags: tags,
+});
+
+const privateSubnet = new aws.ec2.Subnet("private-subnet", {
+  vpcId: vpc.id,
+  cidrBlock: "10.0.1.0/24",
+  tags: tags,
+});
+
+const publicSubnet = new aws.ec2.Subnet("public-subnet", {
+  vpcId: vpc.id,
+  cidrBlock: "10.0.2.0/24",
+  tags: tags,
+});
+
+// Lambda
+const lambdaRole = new aws.iam.Role("info-handler-role", {
+  assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal(
+    aws.iam.Principals.LambdaPrincipal
+  ),
+  managedPolicyArns: [
+    aws.iam.ManagedPolicy.AWSLambdaBasicExecutionRole,
+    aws.iam.ManagedPolicy.AWSLambdaVPCAccessExecutionRole,
+  ],
+  tags: tags,
+});
+
+const infoHandler = new aws.lambda.Function("info-handler", {
+  role: lambdaRole.arn,
+  runtime: aws.lambda.Runtime.Python3d9,
+  code: new pulumi.asset.FileArchive("./lambda/main.py.zip"),
+  handler: "main.info_handler", // fileName.methodName
+  vpcConfig: {
+    subnetIds: [privateSubnet.id],
+    securityGroupIds: [vpc.defaultSecurityGroupId],
+  },
+  tags: tags,
+});
+
+// API Gateway
+const securityGroup = new aws.ec2.SecurityGroup("api-gateway-security-group", {
+  vpcId: vpc.id,
+  ingress: [
+    {
+      description: "HTTP inbound",
+      fromPort: 80,
+      toPort: 80,
+      protocol: "tcp",
+      cidrBlocks: ["0.0.0.0/0"],
+    },
+  ],
+  tags: tags,
+});
+
+const vpcLink = new aws.apigatewayv2.VpcLink("andy-vpc-link", {
+  subnetIds: [privateSubnet.id],
+  securityGroupIds: [securityGroup.id],
+  tags: tags,
+});
+
+const api = new aws.apigatewayv2.Api("andy-api-gateway", {
+  protocolType: "HTTP",
+  tags: tags,
+});
+
+//// https://github.com/pulumi/examples/blob/master/aws-ts-apigatewayv2-http-api/index.ts
+const withApi = new aws.lambda.Permission(
+  "api-lambda-permission",
+  {
+    action: "lambda:InvokeFunction",
+    principal: "apigateway.amazonaws.com",
+    function: infoHandler,
+    sourceArn: pulumi.interpolate`${api.executionArn}/*/*`,
+  },
+  { dependsOn: [api, infoHandler] }
```

Comment:
Is there any reason why you needed the dependOn? pulumi should "discover" it as you are using `infoHandler` and `api.executionArn` in the constructuro of the permission

Reply:
No reason to include it in this particular scenario but I kept it in initially because the example provided by Pulumi had included this property (https://github.com/pulumi/examples/blob/master/aws-ts-apigatewayv2-http-api/index.ts).

From what I've read, I would need it in scenarios where a dependency is not explicitly stated in the resource constructor but is actually needed. Hypothetically, if I wanted my API Gateway to have an `aws.apigatewayv2.Authorizer` resource set up prior to connecting my Lambda to prevent unauthorised access, then I would put the Authorizer resource name in the dependsOn property, is that correct?